### PR TITLE
Csv import

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**

### CODE REVIEW

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco please review and let me know when I'm good to merge (I'll need to rebase first).

Thanks.

**Comment by michaelachrisco** [comment]
Looks like you have an issue with 'sign_in' method on your tests. https://travis-ci.org/westernmilling/entity_master_app/builds/56643985

**Comment by josephbridgwaterrowe** [comment]
Bugger, I forgot to do a blanket rspec.

**Comment by michaelachrisco** [comment]
Taking a look.

File: app/interactors/import_entities.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      location = entity.locations.find_by(:location_name => row['LocationName'])
+
+      create_location(row) if location.nil?
+
+      context.entities << entity
+    end
+  end
+
+  def create_entity(row)
+    entity = Entity.new(:display_name => row['Name'],
```
Comment (michaelachrisco) [inline app/interactors/import_entities.rb]: Any of these rows that have commas as the values will cause the tests to fail. For example: 
```Ruby
  let(:data) do
    [
      %w{Reference Name LocationName Address City Country Region RegionCode},
      [
        Faker::Number.number(8),
        "Hackett Swift and Terry",
        'Default Location',
        Faker::Address.street_address,
        Faker::Address.city,
        'United States',
        Faker::Address.state,
        Faker::Address.zip_code
      ], ...
```
What you will have to do is sanitize the data of the commas either by row or before the import occurs. Or you may need to have the interactor fail if there are commas in the rows.

**Comment by michaelachrisco** [comment]
http://stackoverflow.com/questions/19328613/why-is-the-node-modules-folder-not-committed-to-git

Can you remove the node_modules folder?

File: app/interactors/import_entities.rb (hunk 1 above)
Comment (josephbridgwaterrowe) [inline app/interactors/import_entities.rb]: I believe my last commit fixes this issue.

**Comment by josephbridgwaterrowe** [comment]
One of the links referenced by that item wasn't found and I read somewhere that one should include their node_modules.

**Comment by josephbridgwaterrowe** [comment]
Although some quick checks suggest not, so I will remove :) It looks like there have been some recent changes to the recommendations concerning node_modules.

http://stackoverflow.com/a/19416403/229573

**Comment by michaelachrisco** [comment]
Ok sounds good. It will make the process of looking over the changes a whole lot cleaner.

File: config/application.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+DOMAIN: 'entities.dev'
```
Comment (michaelachrisco) [inline config/application.yml]: :gun: it.

Comment (josephbridgwaterrowe) [inline config/application.yml]: Oh bugger, I'm not sure how the file got in there.

**Comment by michaelachrisco** [comment]
I am going through the csv import branch ATM and setting up for a full test. After this test, I think we are good to rebase and merge.

**Comment by michaelachrisco** [comment]
Ok looks good to me. Import was successful, pages look nice :) Rebase and I will merge.

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco I think we are good to merge now.

**Comment by michaelachrisco** [comment]
Agreed merging.

**Pull request merged**